model/flow/collection_test.go

```diff
@@ -37,6 +37,16 @@ func TestNewCollection(t *testing.T) {
 		assert.NoError(t, err)
 		assert.NotNil(t, col)
 		assert.Len(t, col.Transactions, 1)
+
+		t.Run("convert to LightCollection", func(t *testing.T) {
```

Comment:
Worth checking that it panics in case of converting from invalid transaction.

Reply:
Ending up taking your suggestion to remove the constructor error which eliminates the possibility for a panic from invalid transaction